src/v/storage/disk_log_impl.cc

```diff
@@ -461,29 +461,37 @@ disk_log_impl::monitor_eviction(ss::abort_source& as) {
 
 model::offset
 disk_log_impl::request_eviction_until_offset(model::offset max_offset) {
-    vlog(
-      gclog.debug,
-      "[{}] requested eviction of segments up to {} offset",
-      config().ntp(),
-      max_offset);
-    // we only notify eviction monitor if there are segments to evict
-    auto have_segments_to_evict
-      = (_segs.size() > 1)
-        && _segs.front()->offsets().get_committed_offset() <= max_offset;
+    const auto min_offset = [this]() -> std::optional<model::offset> {
```

Comment:
nit: IMO `min_offset` is kind of confusing, given it isn't clear it's the minimum of what set this is referring to. Same with `max_offset` for that matter

Maybe call it `lowest_removable_offset`, and call `max_offset` `target_offset_to_remove` or something? If you do make this change, would be helpful to also update the log message